cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -399,15 +434,59 @@ class TestNumericMinMaxKernel : public ::testing::Test {
 };
 
 template <typename ArrowType>
-class TestFloatingMinMaxKernel : public TestNumericMinMaxKernel<ArrowType> {};
+class TestBooleanMinMaxKernel : public TestPrimitiveMinMaxKernel<ArrowType> {};
+
+template <typename ArrowType>
+class TestIntegerMinMaxKernel : public TestPrimitiveMinMaxKernel<ArrowType> {};
+
+template <typename ArrowType>
+class TestFloatingMinMaxKernel : public TestPrimitiveMinMaxKernel<ArrowType> {};
+
+TYPED_TEST_SUITE(TestBooleanMinMaxKernel, BooleanArrowType);
```

Comment:
This seems overwrought since there is only one type, can we do it without TYPED_TEST?

Reply:
I chose it in order to reuse `AssertMinMaxIsNull` and `AssertMinMaxIs` methods. 
@bkietz can I instantiate `TestPrimitiveMinMaxKernel` without TYPED_TEST macro?

Reply:
Just define

```
class TestBooleanMinMaxKernel : public TestPrimitiveMinMaxKernel<BooleanType> {};
```

Reply:
Right.